git-annex-remote-rclone

```diff
@@ -216,21 +216,24 @@ while read -r line; do
 			# Any nonzero object count means present.
 			# Some rclone backends support multiple
 			# files under one name.
-			if check_result=$(rclone size "$LOC$key" 2>&1) &&
-				echo "$check_result" | GREP -E 'Total objects: [1-9][0-9]*\>'; then
-				echo CHECKPRESENT-SUCCESS "$key"
+			if check_result=$(rclone size --json "$LOC$key" 2>&1) &&
+			    echo "$check_result" | GREP '"count":'; then
+				if echo "$check_result" | GREP -E '"count":\s*[1-9][0-9]*'; then
+					echo CHECKPRESENT-SUCCESS "$key"
+				else
+					echo CHECKPRESENT-FAILURE "$key"
+				fi
 			else
-				# rclone 1.29 used 'Total objects: 0'
-				# rclone 1.30 uses 'directory not found'
-				if echo "$check_result" | GREP 'Total objects: 0' ||
-				   echo "$check_result" | GREP ' directory not found'; then
+				res="$?"
+				# res now contains the exit code of rclone (left-hand-side of && in if) or grep (right-hand-side of && in if)
+				# GNU grep manual says: the exit status is 0 if a line is selected, 1 if no lines were selected, and 2 if an error occurred.
+				# So for values above 2, res should be the exit code of rclone.
+				# See https://rclone.org/docs/#exit-code for rclone's exit codes.
+				if [[ "$res" == "3" ]] || [[ "$res" == "4" ]]; then
+					# file or directory doesn't exist
```

Comment:
a bit "too many assumptions" to my taste. Why not to break apart that `&&` and have more explicit handling of `$?` after rclone with subsequent  `echo ... | grep ` ? also the former is done multiple times, but could be done once with `sed` to just get the value and react if it is empty or not, e.g. smth like (didn't check) `count=$(sed -n -e 's,"count":\s*\([1-9][0-9]*\),\1,gp`) and then `if [ -n "$count" ]` etc

Reply:
Logic is now hopefully simpler.